app/models/miq_report.rb

```diff
@@ -123,14 +123,31 @@ def self.get_col_info(path)
     }
   end
 
+  # example
+  # converts  "Host.hardware.disks-filename" to
+  # ["Host", "hardware", "disk", "filename"]
+  def self.parse_column(column)
+    prefix = CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX
+    column, column_without_relations = column.split("-#{prefix}") if column.include?(prefix)
+
+    parts = column.split(".")
+
+    if column.include?(prefix)
+      parts.push("#{prefix}#{column_without_relations}")
+    end
+
+    parts[-1] = parts.last.split("-").first # Strip off field name from last element
+
+    parts
+  end
+
   def self.display_filter_details(cols, mode)
     # mode => :field || :tag
     # Only return cols from has_many sub-tables
     return [] if cols.nil?
     cols.inject([]) do |r, c|
       # eg. c = ["Host.Hardware.Disks : File Name", "Host.hardware.disks-filename"]
-      parts = c.last.split(".")
-      parts[-1] = parts.last.split("-").first # Strip off field name from last element
+      parts = parse_column(c.last)
```

Comment:
@zeari I realized one thing, this  method https://github.com/zeari/manageiq/blob/1e8b25e55f9a63114c45e87898906bdd52b8c80c/app/controllers/report_controller/reports/editor.rb#L221

is used just for listing attributes which contains associated attributes which are used as secondary filter option in filter tab of report. (you have to have select any attribute with association for example `Vm.hardware.disks-device_type`)

<img width="900" alt="screen shot 2016-11-18 at 15 01 06" src="https://cloud.githubusercontent.com/assets/14937244/20432535/e1391540-ad9f-11e6-9945-e3385f9a37e4.png">
and we are not support virtual  custom attributes in associations at this moment.
so you can ignore such attribute now (and make this PR simplier) by

`
next if Field.valid_custom_attribute?(c.last) 
`
when this will be merge (https://github.com/ManageIQ/manageiq/pull/12417)